### PR TITLE
DROOLS-3400: Data Types - Validate the Data Type name using the logic that the marshaller implements

### PR DESCRIPTION
**Pull request by karreiro**
See https://issues.jboss.org/browse/DROOLS-3400

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/editors/types/DMNValidationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+package org.kie.workbench.common.dmn.backend.editors.types;
+
+import org.kie.dmn.feel.parser.feel11.FEELParser;
+import org.kie.workbench.common.dmn.api.editors.types.DMNValidationService;
+
+public class DMNValidationServiceImpl implements DMNValidationService {
+
+    @Override
+    public boolean isValidVariableName(final String source) {
+        return FEELParser.checkVariableName(source).isEmpty();
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/editors/types/DMNValidationServiceImpl.java]: I think we should use `FEELParser.isVariableNameValid(source)` as it is implemented exactly as our `DMNValidationServiceImpl.isVariableNameValid(source)`

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/validation/DataTypeNameFormatValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    @Test
+    public void testGetCallbackWhenIsValid() {
+
+        final DataType dataType = mock(DataType.class);
+        final Command onSuccess = mock(Command.class);
+
+        validator.getCallback(dataType, onSuccess).callback(true);
+
+        verify(onSuccess).execute();
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/validation/DataTypeNameFormatValidatorTest.java]: Could you please add assertion: `verify(flashMessageEvent, never()).fire(any(DataTypeFlashMessage.class));`

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/validation/DataTypeNameFormatValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+        final DataType dataType = mock(DataType.class);
+        final Command onSuccess = mock(Command.class);
+        final DataTypeFlashMessage flashMessage = mock(DataTypeFlashMessage.class);
+
+        when(nameIsInvalidErrorMessage.getFlashMessage(dataType)).thenReturn(flashMessage);
+
+        validator.getCallback(dataType, onSuccess).callback(false);
+
+        verify(flashMessageEvent).fire(flashMessage);
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/validation/DataTypeNameFormatValidatorTest.java]: Could you please add assertion: `verify(onSuccess, never()).execute();`

**Comment by karreiro** [comment]
@jomarko @manstis Changes applied. Thank-you :-)

**Comment by manstis** [comment]
@karreiro Please let me know when this is green and I'll merge. Thanks.

**Pull request merged**